staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/types.go

```diff
@@ -53,13 +53,23 @@ type APIServiceSpec struct {
 	// CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
 	CABundle []byte
 
-	// Priority controls the ordering of this API group in the overall discovery document that gets served.
-	// Client tools like `kubectl` use this ordering to derive preference, so this ordering mechanism is important.
-	// Values must be between 1 and 1000
-	// The primary sort is based on priority, ordered lowest number to highest (10 before 20).
+	// GroupPriority controls the ordering of the group with respect to other groups. The lowest of all GroupPriorities
+	// for the same group is used.  Must be greater than zero.
+	// The primary sort is based on GroupPriority, ordered lowest number to highest (10 before 20).
+	// The secondary sort is based on the VersionPriority, ordered lowest number to highest.
+	// The tertiary sort is based on the alphabetical comparison of the name of the object.  (v1.bar before v1.foo)
+	// We'd recommend something like: *.k8s.io (except extensions) at 1000, extensions at 2000
+	// PaaSes (OpenShift, Deis) are recommended to be in the 3000s
+	GroupPriority int32
```

Comment:
> Why not?

* APIService registration is no longer self-contained
* Registration of a single API groupversion would need to create two objects, one of which would get an AlreadyExists error if the group was previously registered
* The status controller has to watch/join two resources, which complicates it

Those are significant enough changes I would not anticipate completing them in the 1.7 timeframe. I also don't think the benefits outweigh the extra complexity of a separate resource representing the API group.

The priority field as it is now is overloaded to mean two things. The first use case we had for this field (preserving existing order of underlying groups, and versions within a group from the delegated server) found it difficult to use.

We should make simple change to specify the distinct pieces of information required.